src/Sylius/Bundle/UserBundle/Resources/config/validation.xml

```diff
@@ -52,6 +52,12 @@
                 <option name="groups">sylius_customer_profile</option>
             </constraint>
         </property>
+        <property name="gender">
+            <constraint name="NotBlank">
+                <option name="message">sylius.customer.gender.not_blank</option>
+                <option name="groups">sylius_customer_profile</option>
```

Comment:
We should use `sylius` group since this field is mandatory in the database, so we should always validate that.